inst/tests/tests.Rraw

```diff
@@ -21079,3 +21079,8 @@ setindex(DT, b)
 # make sure that print(DT) doesn't warn due to the header missing index column types, #6806
 # can't use output= here because the print() call is outside withCallingHandlers(...)
 test(2307, { capture.output(print(DT, class = TRUE, show.indices = TRUE)); TRUE })
+
+# test for enhancing error message of invalid column #6512
+test(2308, {
+  melt(data.table(A = 1:5, B = 6:10), id.vars = c("A", "-1"))
+}, error = "One or more values in 'id.vars' are invalid")
```

Comment:
this test case is irrelevant.

please add test cases for the new functionality.